speeduino/decoders.ino

```diff
@@ -427,14 +431,15 @@ void triggerSetup_missingTooth(void)
     BIT_SET(decoderState, BIT_DECODER_IS_SEQUENTIAL);
   } 
   triggerActualTeeth = configPage4.triggerTeeth - configPage4.triggerMissingTeeth; //The number of physical teeth on the wheel. Doing this here saves us a calculation each time in the interrupt
-  triggerFilterTime = (1000000 / (MAX_RPM / 60 * configPage4.triggerTeeth)); //Trigger filter time is the shortest possible time (in uS) that there can be between crank teeth (ie at max RPM). Any pulses that occur faster than this time will be discarded as noise
+  triggerFilterTime = ((1000000 / (MAX_RPM / 60 * configPage4.triggerTeeth))/2); //Trigger filter time is the shortest possible time (in uS) that there can be between crank teeth (ie at max RPM). Any pulses that occur faster than this time will be discarded as noise
   if (configPage4.trigPatternSec == SEC_TRIGGER_4_1)
   {
     triggerSecFilterTime = 1000000 * 60 / MAX_RPM / 4 / 2;
   }
   else 
   {
-    triggerSecFilterTime = (1000000 / (MAX_RPM / 60));
+    //triggerSecFilterTime = (1000000 / (MAX_RPM / 60)/2); //this is too large, the single pulse on the secondary may be quite small
+    triggerSecFilterTime=triggerFilterTime;
```

Comment:
Yeah I see:  The secondary filter time should not be calculated using the primary teeth count!

Duration of the pulse is relevant with the new filter because it is designed to filter out short noise pulses. So care must be taken here not to filter out the correct pulses too! So the noise filter value can be calculated from the minimum tooth width. To be determined...